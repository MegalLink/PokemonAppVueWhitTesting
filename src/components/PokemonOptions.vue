<template>
  <div class="container-options">
    <ul>
      <li
        v-for="pokemon in pokemons"
        :key="pokemon.id"
        @click="$emit('selectedPokemon', pokemon.id)"
      >
        {{ pokemon.name }}
      </li>
    </ul>
  </div>
</template>
<script>
export default {
  //LA emicion es un array de datos [pokemon.id, otherValue] si mandamos mas datos seria con ,
  props: {
    pokemons: {
      type: Array,
      required: true,
    },
  },
};
</script>
<style scoped>
.container-options {
  display: flex;
  justify-content: center;
}
/* Pokemon Options */
ul {
  list-style-type: none;
}
li {
  background-color: white;
  border-radius: 5px;
  border: 1px solid rgba(0, 0, 0, 0.2);
  cursor: pointer;
  margin-bottom: 10px;
  width: 250px;
}

li:hover {
  background-color: rgba(0, 0, 0, 0.05);
}
</style>
