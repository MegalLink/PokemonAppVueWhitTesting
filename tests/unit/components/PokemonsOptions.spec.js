import { shallowMount } from "@vue/test-utils";
import PokemonOptions from "@/components/PokemonOptions";
import pokemonsMock from "../mocks/pokemons.mock";

describe("PokemonPicture Component Tests", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(PokemonOptions, {
      props: { pokemons: pokemonsMock },
    });
  });
  it("Should match snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Should match li with information pokemon with array data", () => {
    const li = wrapper.findAll("li");
    expect(li.length).toBe(4);
    for (let i = 0; i < li.length; i++) {
      const liPokemonName = li[i].text();
      const liId = li[i].attributes("key");
      expect(liPokemonName == pokemonsMock[i].name);
      expect(liId == pokemonsMock[i].id);
    }
  });

  it("Should emmit id from li selected", () => {
    const firstLi = wrapper.find("li");
    const li = wrapper.findAll("li");
    firstLi.trigger("click");
    expect(wrapper.emitted("selectedPokemon")).toEqual([[pokemonsMock[0].id]]);
    expect(li.length).toBe(4);
  });
});
