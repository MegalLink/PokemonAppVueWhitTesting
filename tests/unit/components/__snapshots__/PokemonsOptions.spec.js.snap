// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture Component Tests Should match snapshot 1`] = `
<div class="container-options">
  <ul>
    <li>pikachu</li>
    <li>sevaiper</li>
    <li>ponyta</li>
    <li>huevofet</li>
  </ul>
</div>
`;
